src/code/views/modal-tabbed-dialog-view.coffee

```diff
@@ -0,0 +1,43 @@
+ModalDialog = React.createFactory require './modal-dialog-view'
+{div, ul, li, a} = React.DOM
+
+class TabInfo
+  constructor: (settings={}) ->
+    {@label, @component} = settings
+
+Tab = React.createFactory React.createClass
+
+  displayName: 'ModalTabbedDialogTab'
+
+  clicked: (e) ->
+    e.preventDefault()
+    @props.onSelected @props.index
+    
+  render: ->
+    (li {className: (if @props.selected then 'tab-selected' else ''), onClick: @clicked}, @props.label)
+
+module.exports = React.createClass
+
+  displayName: 'ModalTabbedDialog'
+
+  getInitialState: ->
+    selectedTabIndex: 0
+
+  statics:
```

Comment:
I probably shouldn't have overloaded the word 'Tab'.  The render method uses the Tab from line 8.  The Tab on line 27 lets us export the TabInfo class.